src/mongocxx/client_session.hpp

```diff
@@ -90,8 +98,23 @@ class MONGOCXX_API client_session {
     ///
     /// Get the server_id the session is pinned to. The server_id is zero if the session is not
     /// pinned to a server.
+    ///
     std::uint32_t server_id() const noexcept;
 
+    ///
+    /// Indicates whether session has been marked “dirty” as defined in the driver sessions
+    /// specification.
+    ///
+    /// @see
+    /// https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.rst
+    ///
+    bool get_dirty() const noexcept;
+
+    ///
+    /// Returns the current transaction state for this session.
+    ///
+    transaction_state get_transaction_state() const noexcept;
```

Comment:
These were taken from libmongoc's public API. `get_transaction_state` seems like it should be public but `get_dirty` was only exposed in libmongoc for wrapping drivers I believe. I could try to see if we could keep `get_dirty` private.

Reply:
Ah, I missed this comment on first review. While I agree that `get_dirty` does not seem useful to users, I don't see a great alternative for accessing the underlying `mongoc_client_session_t` from tests. These were a couple alternatives I thought of:

1. Keep this public but mark it with the `MONGOCXX_PRIVATE` macro. There is some precedent for that in the APM events:
https://github.com/mongodb/mongo-cxx-driver/blob/a46914ade52a1fb63e79207250458c81da513bf0/src/mongocxx/events/heartbeat_started_event.hpp#L35:L35 That marks it as "linker private" so consumers cannot actually link to the method from their code. However, that also applies to the test executable, so that doesn't work.
2. Mark this as a friend of a class that the test runner can utilize. But including test classes in public headers seems like a dealbreaker.

`get_dirty` isn't useful to users, but if we have no reasonable alternative, then so be it. Let's keep it as-is.